pygmt/helpers/testing.py

```diff
@@ -139,3 +140,41 @@ def wrapper(*args, ext="png", request=None, **kwargs):
         return wrapper
 
     return decorator
+
+
+def download_test_data():
+    """
+    Convenient function to download data files used in tests.
+    """
+    # List of earth_relief suffix to download
+    earth_relief_suffix = [
+        "01d_p",
+        "01d_g",
+        "30m_p",
+        "30m_g",
+        "10m_p",
+        "10m_g",
+        "05m_p",
+        "05m_g",
+    ]
+    # List of tiles of 03s/01s srtm data.
+    # Names like @N35E135.earth_relief_03s_g.nc is for internal use only.
+    # The naming scheme may change. DO NOT USE IT IN YOUR SCRIPTS.
+    earth_relief_tiles = ["@N35E135.earth_relief_03s_g.nc"]
+    # List of cache files
+    cache_files = [
+        "@fractures_06.txt",
+        "@ridge.txt",
+        "@srtm_tiles.nc",  # needed for 03s and 01s relief data
+        "@Table_5_11.txt",
+        "@test.dat.nc",
+        "@tut_bathy.nc",
+        "@tut_quakes.ngdc",
+        "@tut_ship.xyz",
+        "@usgs_quakes_22.txt",
+    ]
+
+    dataset = [f"@earth_relief_{suffix}" for suffix in earth_relief_suffix]
+    dataset.extend(earth_relief_tiles + cache_files)
```

Comment:
A few questions/comments on this:

1. Is there a need to split the datasets up into different lists? Or could we just put them all in a single list since we're downloading all of them anyway (and avoid using `list.extend()`). Inline comments can still be used.
2. Will there be a unit test added for this function?
3. Not sure how English works, but should the variable 'dataset' be 'datasets' (plural)? I know data is singular/plural, but not sure about dataset :shrug:

Reply:
> 1. Is there a need to split the datasets up into different lists? Or could we just put them all in a single list since we're downloading all of them anyway (and avoid using `list.extend()`). Inline comments can still be used.

I'm OK with writing all files in a single list.

> 2. Will there be a unit test added for this function?

I'm not sure if a unit test really helps. All these files should have been downloaded by the `cache_data.yml` workflow, the unit test only won't download these files again.

> 3. Not sure how English works, but should the variable 'dataset' be 'datasets' (plural)? I know data is already plural, but not sure about dataset 🤷

I think it should be "datasets".

Reply:
> > 1. Will there be a unit test added for this function?
> 
> I'm not sure if a unit test really helps. All these files should have been downloaded by the `cache_data.yml` workflow, the unit test only won't download these files again.

I guess it's ok not having a test. Just asking because these lines of code are marked as not tested by codecov, but we can ignore that :slightly_smiling_face: